starlette/websockets.py

```diff
@@ -181,6 +202,12 @@ async def close(
             {"type": "websocket.close", "code": code, "reason": reason or ""}
         )
 
+    async def send_response(self, response: Response) -> None:
+        if self._have_response_extension():
+            await response(self.scope, self.receive, self.send)
+        else:
+            await self.close(code=1008, reason=f"HTTP Response {response.status_code}")
```

Comment:
My thinking was that the application shouldn't have to have code dealing with server peculiarities. The application could simply always create an error response, and leave it to Starlette to decide what to do with it.  If the server supports it, then fine.  If not, Starlette automatically does the next best thing, simply close the socket, which is specified to create a 403 response.   
The fallback code and message are arbitrary, because ASGI does not specify what should happen to these codes in case of a 403.  _maybe_ the server adds that to the 403 payload, maybe not.  I haven't looked at what uvicorn does.

Reply:
If this is the path we want to go, then let's remove the code and reason. Both doesn't make sense, since the handshake didn't happen, and the server actually should ignore both code and reason.

Reply:
I'm still not sure if we want this... If you are using `send_response`, I'd expect that working out, and if it doesn't, as a developer I'd like to see an exception :thinking:

Reply:
But that exception appearing or not depends on which server you choose to run your app.  The response here is informative only.  Starlette is ASGI toolkit aimed at simplifying development.
As I see it, here are the options:
1) just close the connection().  This is always supported and is the original way of rejecting a connection.  Server provides a 403 response.
2) Close the response, but proide additional information for those interested.
  2a) Server supports the extension, and a response body is delivered along with the message.
  2b) Server does not support the response and body is ignored.
  
It is th case 2b) which worries you.  IMHO, the response body is informative only.  Depending on web server a response body can or can not be delivered.   If the information is somehow protocol critical (not recommended), then the application can always check for the presence of the extension.

Ok, how about this:  Instead of an exception, how about a warning?  This will get flagged to the developer but it won't crash your application if an extension isn't supported.   And we won't have to push the burden on the developer to always check the state for the presence of the extension before using a close call with an extended api?

Reply:
I understand what you are saying, but I don't think the premise is right.

If you start using `send_denial_response`, and you don't have an error, you'll believe it's working as expected, when it actually is not.

Reply:
My thinking is that the denial response is an informative event.  It augments the plain close event, possibly with
some nicely formatted data, information about what is missing, etc.  But it is not essential to the operation of the _service_.

I understand it as "I'm not accepting this connection.  Here is some additional information to the client, _if_ I can send it and _if_ he wants to receive it, but otherwise, I'm done with the connection."

This is why I suggested that maybe a warning were appropriate.

But I don't feel strongly about this anymore and am happy to do it either way, so having made my point, I can make it an exception.